### ISSUE TITLE
Removing Certain rail station tile crashes game

### ISSUE THREAD

**Issue by superchadST345** [OpenLoco/OpenLoco#2357]
**Version information:**
 - Windows 11
 - 24.02.1

**Describe the bug**
Not sure what causes it, but removing some station tiles (for me, it was a very specific station tile next to a road/rail intersection

crash dump uploaded

[crashdump.zip](https://github.com/OpenLoco/OpenLoco/files/14672789/crashdump.zip)

**To Reproduce**
1. build road and rail intersection
2. place station on track next to it
3. remove station
4. game may crash

**Expected behaviour**
game should never crash removing a station

**Comment by AaronVanGeffen** [comment]
We've not reimplemented the station removal game command yet, so it might be hard to narrow down. Hopefully the dump will reveal what's happening, but in case it isn't: could you provide a save game and a screenshot, pointing out what tile needs removing to trigger the crash?

**Comment by superchadST345** [comment]
attached screenshot and save game

![image](https://github.com/OpenLoco/OpenLoco/assets/6529196/1ccce79c-2582-4716-bbcc-403d964186a7)
[Sandbox Settler-152.zip](https://github.com/OpenLoco/OpenLoco/files/14672946/Sandbox.Settler-152.zip)

**Comment by superchadST345** [comment]
trying to remove the lone station piece

**Comment by duncanspumpkin** [comment]
This uses custom objects airport4 and hport2 do you have a copy of them you could post.

**Comment by superchadST345** [comment]
I know the airport should be here https://www.owenrudge.net/download/download?t=10&id=225 and the slopes should here https://www.wisim-welt.de/filebase/index.php?entry/1028-default-bridges-w-new-slopes-added/

**Comment by duncanspumpkin** [comment]
Its crashing because the station tile hasn't been added to the station. Which is odd! Most likely this is a mistake in the train station placement code (that was done last month).

**Comment by superchadST345** [comment]
so does it have a null station then?

**Comment by superchadST345** [comment]
If i click on it, it opens the station

**Comment by AaronVanGeffen** [comment]
Station tile elements know what station they belong to, but stations themselves also keep track of what tile elements belong to the station. Sounds like the former has been registered correctly, but the latter hasn't.

**Comment by superchadST345** [comment]
interesting, is there a way to catch the exception and remove the station rather than crashing the game?

**Comment by superchadST345** [comment]
where is the code to add the station at? where is the code to remove the station at?

**Comment by AaronVanGeffen** [comment]
> where is the code to add the station at?

[CreateTrainStation.cpp](https://github.com/OpenLoco/OpenLoco/blob/master/src/OpenLoco/src/GameCommands/Track/CreateTrainStation.cpp)

> where is the code to remove the station at?

This has not been reimplemented in C++ yet.

**Comment by NSTrains** [comment]
I'm still getting this issue on 24.03

**Comment by duncanspumpkin** [comment]
@superchadST345 did you happen to build this station up to 80+ tiles then start removing them (in a different order to the way they were placed)? Also @NSTrains did you build a station that was 80+ tiles when the crashing started? I think #2396 will fix this but i'm not too sure as the save in this issue has only 64 tiles but maybe it originally had 80+ but it had a number of tiles then removed without issue.

**Comment by NSTrains** [comment]
@duncanspumpkin its both crashing without me doing anything and if build a station tile then remove it, it crashes too

**Comment by superchadST345** [comment]
@duncanspumpkin honestly I can't remember, it was a large station but i don't know how large, and yes I did remove them in a different order than they where created

also right now on a station that won't let me add another tile due to being too large, if i remove a station, add it back, and remove it, it crashes

**Comment by duncanspumpkin** [comment]
Okays I'm pretty sure it's all fixed by #2396. Unfortunately can't fix saves that were made during the time this bug was happening. I.e. 24.02 - 24.03.

**Comment by superchadST345** [comment]
@duncanspumpkin is there anything that can be done to fix those saves down the road? isn't the bug caused by a station not knowing that a tile belongs to it, couldn't something be added to remove the tile anyways if it isn't properly associated? or at the very least create an exception catching that prevents the removal of a station to at least prevent a game crash

**Comment by duncanspumpkin** [comment]
> @duncanspumpkin is there anything that can be done to fix those saves down the road? isn't the bug caused by a station not knowing that a tile belongs to it, couldn't something be added to remove the tile anyways if it isn't properly associated? or at the very least create an exception catching that prevents the removal of a station to at least prevent a game crash

We haven't implemented train station removal so we can't easily prevent the crash. Once we have implemented it we could maybe get it to work.

**Comment by NSTrains** [comment]
24.04 still has the same problem, it is most likely the save game that I'm using.

**Comment by superchadST345** [comment]
@duncanspumpkin is there any progress on station removal?

**Comment by AaronVanGeffen** [comment]
Could you try the experimental build from [this page](https://github.com/OpenLoco/OpenLoco/actions/runs/8620485090?pr=2410#artifacts)? Look for 'Artifacts' at the bottom.

This build from #2410 implements the removeTrainStation command, and takes overly large stations like yours into account. However, it is experimental, so please keep a backup of your save file, just in case.

**Comment by superchadST345** [comment]
@AaronVanGeffen does it take into account the original problem for this issue of the station not being added to the station? the original issue was the station tile knowing what station it is part of, but not the station knowing what station tile is part of it at least according to @duncanspumpkin

**Comment by AaronVanGeffen** [comment]
Yes, my implementation should be more flexible than vanilla. Please try it and report your findings.
